client/src/Main/Monitor/MonitorReducers.js

```diff
@@ -212,27 +212,45 @@ const MonitorReducers = (state = currentInitialState, action) => {
   case UPDATE:
     newState = action.state;
     break;
-  case SET_CURRENT_TASK_FIELDS: {
-    let currentTask = state.tasks[state.currentTaskIndex];
-    currentTask = {
-      ...currentTask,
-      ...action.fields,
-    };
+  case SET_TASK_FIELDS:
+    {
+      let currentTask = state.tasks[action.taskIndex];
+      currentTask = {
+        ...currentTask,
+        ...action.fields,
+      };
 
-    const tasks = state.tasks;
-    tasks[state.currentTaskIndex] = currentTask;
+      const tasks = state.tasks;
+      tasks[action.taskIndex] = currentTask;
 
-    newState = {
-      ...state,
-      tasks,
-    };
-  }
+      newState = {
+        ...state,
+        tasks,
+      };
+    }
+    break;
+  case SET_CURRENT_TASK_FIELDS:
+    {
+      let currentTask = state.tasks[state.currentTaskIndex];
+      currentTask = {
+        ...currentTask,
+        ...action.fields,
+      };
+
+      const tasks = state.tasks;
+      tasks[state.currentTaskIndex] = currentTask;
+
+      newState = {
+        ...state,
+        tasks,
+      };
```

Comment:
This logic is duplicated, can you add an helper fonction on top of this file
```suggestion
      const updateTask = (state, taskIndex, newFields) => {
        let taskToUpdate = state.tasks[taskIndex];
        taskToUpdate = {
          ...taskToUpdate,
          ...action.fields,
        };

        const tasks = state.tasks;
        tasks[taskIndex] = taskToUpdate;

        return {
          ...state,
          tasks,
        };
      } 
```
And then use it in each case :
```suggestion
case SET_TASK_FIELDS:
  {
      newState = updateTask(state, action.taskIndex, action.fields);
  }
case SET_CURRENT_TASK_FIELDS:
  {
      newState = updateTask(state, state.currentTaskIndex, action.fields);
  }
```

Reply:
👍 Done :)